src/main/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsWriter.java

```diff
@@ -134,6 +145,7 @@ public void finish() throws IOException {
      */
     @Override
     public void close() throws IOException {
+        quantizationStateWriter.closeOutput();
```

Comment:
needs a null check

Reply:
This is making me think why ITs are not failing, as QSW will be null for nmslib and faiss engine HNSW

Reply:
Added